src/core/execute.c

```diff
@@ -2360,10 +2361,14 @@ static int setup_exec_directory(
                                         goto fail;
                         }
 
-                        /* And link it up from the original place */
-                        r = symlink_idempotent(pp, p, true);
-                        if (r < 0)
-                                goto fail;
+                        /* And link it up from the original place, unless we are going to have a
+                         * mount namespace, in which case this happens later to allow configuring
+                         * empty var/run/etc. */
+                        if (!needs_mount_namespace) {
```

Comment:
hmm, without this conditionalization things would work too, why have it? can you elaborate?

Reply:
In case there's no TempFS it would symlink twice (as it's done again later), and in case there is it creates a pointless symlink on the host filesystem that is not used. It seemed cleaner to avoid it, since it's easy to check.

Reply:
is it pointless on the host? i mean the admin might want to access the dir under its non-private name, too?